# HG changeset patch
# User Joe Rothweiler <octaveuser@sensicomm.com>
# Date 1246425523 -7200
#      Wed Jul 01 07:18:43 2009 +0200
# Node ID c9636d98e5cd785f3cada2c05c4ef4421e635306
# Parent  2ad67ccd09fd9103b75624d8550892b3bac31a95
fix kbhit(1) to not discard waiting keystrokes

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-07-01  Joe Rothweiler <octaveuser@sensicomm.com>
+
+	* input.cc (raw_mode): Use TCSADRAIN if no wait.
+
 2009-06-30  Jaroslav Hajek  <highegg@gmail.com>
 
 	* symtab.h (force_variable): Remove assertion.
 
 2009-06-28  Michael Goffioul  <michael.goffioul@gmail.com>
 
 	* Makefile.in (INCLUDES): Install txt-eng.h and txt-eng-ft.h.
 	* gl-render.cc (opengl_renderer::draw_text): Fix bitmap offset
diff --git a/src/sysdep.cc b/src/sysdep.cc
--- a/src/sysdep.cc
+++ b/src/sysdep.cc
@@ -387,17 +387,17 @@ raw_mode (bool on, bool wait)
       }      
     else
       {
 	// Restore saved modes.
 
 	s = save_term;
       }
 
-    tcsetattr (tty_fd, TCSAFLUSH, &s);
+    tcsetattr (tty_fd, wait ? TCSAFLUSH : TCSADRAIN, &s);
   }
 #elif defined (HAVE_TERMIO_H)
   {
     struct termio s;
     static struct termio save_term;
 
     if (on)
       {
